app/pages/admin/project-status.jsx

```diff
@@ -76,10 +79,28 @@ class ProjectStatus extends Component {
   handleToggle(event, workflow) {
     this.setState({ error: null });
     const checked = event.target.checked;
+    const defaultWorkflowId = this.state.project.configuration.default_workflow;
+
+    const promises = [this.getWorkflows(), this.getProject()];
+
+    if (defaultWorkflowId === workflow.id && workflow.active) {
+      Dialog.alert(
+        <WorkflowDefaultDialog closeButton={true} required={true} />
+      )
+      .then(() => {
+        Promise.all(promises)
```

Comment:
This is related to your question in the PR: 
> if a user clicks the background or presses escape, the workflow retains its default status, but workflow.active still gets toggled (ie, it is still possible to have a default workflow that is inactive). Not sure if this is the desired UX, would appreciate input.

My suggestion to use `Promise.all()` was for the project _update_ and the workflow _update_. If a workflow is active and the project default, we want to prompt the admin user asking to confirm the switch and if they confirm, then remove that workflow as the default and make it inactive. If they cancel, then no action is taken.

At this point, you shouldn't first need to do a get request for the project and workflows again as you should already have them since they are called in `componentDidMount`. The get request for the project and workflows should be after the successful `Promise.all()`. What's defined in `componentDidMount` could be moved into a class method to be reused here as well as in `componentDidMount`.